### PR TITLE
Remove several methods from App interface

### PR DESCRIPTION
**Pull request by pchico83**
This is just a cleanup of the App interface refactor

### CODE REVIEW

File: cmd/up/up.go [hunk 1]
```diff
@@ -327,9 +327,6 @@ func (up *upContext) getApp(ctx context.Context) (apps.App, bool, error) {
 		return apps.NewDeploymentApp(apps.GetDeploymentSandbox(up.Dev)), true, nil
 	}
 	if err == nil {
-		if app.ObjectMeta().Annotations[model.OktetoAutoCreateAnnotation] != model.OktetoUpCmd {
-			up.isSwap = true
```
Comment (ifbyol) [inline cmd/up/up.go:331]: Just curiosity, what was the meaning of this flag?

File: pkg/k8s/apps/crud.go [hunk 2]
```diff
@@ -50,6 +50,41 @@ func Get(ctx context.Context, dev *model.Dev, namespace string, c kubernetes.Int
 	return &StatefulSetApp{sfs: sfs}, nil
 }
 
+//IsDevModeOn returns if a statefulset is in devmode
+func IsDevModeOn(app App) bool {
+	return app.ObjectMeta().Labels[model.DevLabel] == "true"
+}
+
+func (t *Translation) DevModeOff() {
+	t.App.DevModeOff(t)
```
Comment (ifbyol) [inline pkg/k8s/apps/crud.go:59]: This generates a weird circular dependency. If `App.DevModeOff` would accidentally call to `t.DevModeOff()`, we would have an infinite loop. Is there a way we could avoid it? Would it make more complex the code?

**Review by ifbyol — APPROVED** [review]

File: cmd/up/up.go (hunk 1 above)
Comment (pchico83) [reply cmd/up/up.go:331]: it was related to `autocreate`, before existing the `autocreate` field

File: pkg/k8s/apps/crud.go (hunk 2 above)
Comment (pchico83) [reply pkg/k8s/apps/crud.go:59]: I don't like it either :-)
But next refactor will eliminate the object `Translation` as it is, I think it is better to fix it on that iteration

Comment (ifbyol) [reply pkg/k8s/apps/crud.go:59]: Sounds good!

**Pull request merged**